ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala

```diff
@@ -700,8 +675,227 @@ object LedgerApiErrors extends LedgerApiErrorGroup {
+          //                   If the key is big, it can force chunking other resources.
+          (ErrorResource.ContractKey, _key.toString())
+        )
+      }
+
+      case class LedgerReject(override val cause: String)(implicit
+          loggingContext: ContextualizedErrorLogger
+      ) extends LoggingTransactionErrorImpl(cause = cause)
+    }
+
+    @Explanation("An invalid transaction submission was not detected by the participant.")
+    @Resolution("Contact support.")
+    @deprecated("Corresponds to transaction submission rejections that are not produced anymore.")
+    @DeprecatedDocs(
+      "Corresponds to transaction submission rejections that are not produced anymore."
+    )
+    object Disputed
+        extends ErrorCode(
+          id = "DISPUTED",
+          ErrorCategory.SystemInternalAssumptionViolated, // It should have been caught by the participant
+        ) {
+      case class Reject(
+          details: String
+      )(implicit loggingContext: ContextualizedErrorLogger)
+          extends LoggingTransactionErrorImpl(
+            cause = s"Disputed: $details"
+          )
+    }
+
+    @Explanation(
+      "The Participant node did not have sufficient resource quota to submit the transaction."
+    )
+    @Resolution("Inspect the error message and retry after after correcting the underlying issue.")
+    @deprecated("Corresponds to transaction submission rejections that are not produced anymore.")
+    @DeprecatedDocs(
+      "Corresponds to transaction submission rejections that are not produced anymore."
+    )
+    object OutOfQuota
+        extends ErrorCode(id = "OUT_OF_QUOTA", ErrorCategory.ContentionOnSharedResources) {
+      case class Reject(reason: String)(implicit
+          loggingContext: ContextualizedErrorLogger
+      ) extends LoggingTransactionErrorImpl(cause = reason)
+    }
+
+    @Explanation("A submitting party is not authorized to act through the participant.")
+    @Resolution("Contact the participant operator or re-submit with an authorized party.")
+    object SubmitterCannotActViaParticipant
+        extends ErrorCode(
+          id = "SUBMITTER_CANNOT_ACT_VIA_PARTICIPANT",
+          ErrorCategory.InsufficientPermission,
+        ) {
+      case class Reject(
+          details: String,
+          submitter: String = "N/A",
+          participantId: String = "N/A",
+      )(implicit loggingContext: ContextualizedErrorLogger)
+          extends LoggingTransactionErrorImpl(cause = s"Inconsistent: $details")
+    }
+
+    @Explanation(
+      "The ledger time of the submission violated some constraint on the ledger time."
+    )
+    @Resolution("Retry the transaction submission.")
+    object InvalidLedgerTime
+        extends ErrorCode(
+          id = "INVALID_LEDGER_TIME",
+          ErrorCategory.InvalidGivenCurrentSystemStateOther, // It may succeed at a later time
+        ) {
+      case class RejectEnriched(
+          details: String,
+          ledger_time: Instant,
```

Comment:
Please don't use underscore in argument & variable names as this does not conform with the [Scala style guide](https://docs.scala-lang.org/style/naming-conventions.html).

Reply:
Those args are copied from another class. I'll change 👍

Reply:
Actually, these args name make it as metadata to the error details, where the standard is snake_case in the original KVErrors PR (see [here](https://github.com/digital-asset/daml/pull/11326/files#diff-6706187b0e7b98fbc6cde016d257048af0b510eca4403cda4c47d6b10e57698aR295-R297)). If that's the standard for KVErrors, than this change will just stand out. Leaving this as is in this PR, since the change will involve revisiting the other error codes as well (for consistency)

Reply:
Yes, please leave these fields in snake_case.

Reply:
We must revisit this approach in a separate PR as the current solution is 1) not transparent and 2) incompatible with the style guide. As a side-effect of using snake_case you get inclusion into the error metadata which should be done in a more explicit way (or you should need to specifically exclude certain fields to be *not* added, depending on the usage patterns).
Instead of relying on snake_case, there are less obscure methods for achieving the latter, such as using annotations or just explicitly declaring what the context should contain using an `override`.